core/txpool/legacypool/celo_list.go

```diff
@@ -0,0 +1,248 @@
+package legacypool
+
+import (
+	"math"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/txpool"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type celo_list struct {
+	list      *list
+	totalCost map[common.Address]*big.Int
```

Comment:
Have you thought about just making this change in the `list` struct? Maybe we wouldn't need the additional `celo_list` then? And some changes are required anyways because you have to adapt usage to `TotalCost`.